File: python_example/my_strlen.py
##
## EPITECH PROJECT, 2025
## thonny
## File description:
## my_strlen
##

from utils import my_putnbr

def my_strlen(str):
    length = 0

    while str[length] != '\0':
        length += 1
    return length

def main():
    str = "Hello, World!\0"
    length = my_strlen(str)

    my_putnbr(length)
    return 0

main()

File: c_example/my_strlen.c
/*
** EPITECH PROJECT, 2025
** thonny
** File description:
** my_strlen
*/

void my_putnbr(int nb);

int my_strlen(char const *str)
{
    int i = 0;

    while (str[i] != '\0') {
        i++;
    }
    return (i);
}

int main(void)
{
    char str[] = "Hello, World!";

    my_putnbr(my_strlen(str));
    return (0);
}
